python/hsfs/training_dataset.py

```diff
@@ -248,7 +250,22 @@ def compute_statistics(self):
         feature store.
         """
         if self.statistics_config.enabled and engine.get_type() == "spark":
-            return self._statistics_engine.compute_statistics(self, self.read())
+            if self.splits:
+                statistics_of_splits = []
+                for split_name in self.splits:
+                    statistics_of_splits.append(
+                        split_statistics.SplitStatistics(
+                            split_name,
+                            self._statistics_engine.profile_statistics(
+                                self, self.read(split_name)
+                            ),
+                        )
+                    )
+                return self._statistics_engine.register_split_statistics(
+                    self, statistics_of_splits
+                )
+            else:
+                return self._statistics_engine.compute_statistics(self, self.read())
```

Comment:
Same comment as for the Java part. Let's be consistent with our policy of having all the business logic in the engines.